[Wang Jingjing] iP

potaotototo opened the pull request:
# Noosy Project Pull Request

## Features Implemented:

- Task management with todo, deadline, and event commands
- Search tasks by keyword or date
- Mark and unmark tasks as completed
- Delete tasks from the list
- Save tasks to file and load them automatically upon startup

## How to Use Noosy:

- Add a task using the todo, deadline, or event commands.
- Use list to view all tasks.
- Use mark [task number] to mark a task as complete.
- Delete tasks using delete [task number].

Sample Command (with Syntax Highlighting):
```
// Adding a new task with a deadline
deadline submit report /by 2024-09-25
```

Next Steps:
- [ ] Implement file handling for task persistence.
- [ ]  Add date/time parsing for deadlines and events.
- [ ]  Implement reminders for upcoming deadlines.


Noosy is always ready to help you stay organized! ✨

> "The best way to predict the future is to create it." — Abraham Lincoln

You can find the full documentation for Noosy [here](https://github.com/potaotototo/ip/).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

potaotototo committed:
Completed Level 0 – added greeting

potaotototo committed:
Completed Level 1 – echo

potaotototo committed:
Completed Level 2 – list

potaotototo committed:
Completed Level 3 – mark

potaotototo committed:
Completed Level-4 – different tasks

potaotototo committed:
Completed Level 5 – error messages

potaotototo committed:
Completed Level 6 – delete

potaotototo committed:
Completed Level 6 – delete

potaotototo committed:
Completed Level 7 – save tasks

potaotototo committed:
Merge Level 7 branch into master

potaotototo committed:
Update README.md

potaotototo committed:
Complete Level 8 – date and time

potaotototo committed:
Merge branch-Level-8 into master

potaotototo committed:
Merge branch 'master' of https://github.com/potaotototo/ip
Merge into master

potaotototo committed:
Add new classes

potaotototo committed:
Complete A-More-OOP

potaotototo committed:
Complete A-MoreOOP

potaotototo committed:
Merge branch branch-A-MoreOOP into master

slye20 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    // get task name
+    public String getDesc() {
```
"Desc" might sound rather slangy. Maybe getDescription instead?

slye20 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+    @Override
+    public String storeInFile() {
+        String formattedDate = (due != null) ? due.format(OUTPUT_FORMATTER) : "Date can't be processed by Noosy";
+        return String.format("T | %s | %s | %s", super.storeInFile(), this.name, this.due);
+    }
```
Would a more descriptive name like toFileString be better?

slye20 commented on src/main/java/OnCommand.java:
```diff
@@ -0,0 +1,36 @@
+
+public class OnCommand extends Command {
+
+    private LocalDate date;
```
Is date refering to the start or end date?

slye20 reviewed (commented):
LGTM! Just some minor improvements!

potaotototo committed:
Add packages

potaotototo committed:
Merge branch 'branch-A-Packages' into master

potaotototo committed:
Merge branch 'origin/add-gradle-support' into master

rubinnn commented on src/main/java/Noosy.java:
```diff
@@ -0,0 +1,49 @@
+import noosy.ui.Ui;
+import noosy.command.Command;
+import noosy.exception.NoosyException;
+import noosy.task.TaskList;
```
Would be great if you arranged it in sequential order.

rubinnn commented on src/main/java/noosy/parser/Parser.java:
```diff
@@ -0,0 +1,121 @@
+        String firstWord = separated[0];
+        String input = separated.length > 1 ? separated[1] : null;
+
+        switch (firstWord) {
```
Good use of switch case. Indentation may be improved

rubinnn commented on src/main/java/noosy/parser/Parser.java:
```diff
@@ -0,0 +1,121 @@
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
```
Good job in grouping the imports

rubinnn commented on src/main/java/noosy/task/Task.java:
```diff
@@ -0,0 +1,45 @@
+package noosy.task;
```
Easy to read code.Good Job

rubinnn commented on src/main/java/noosy/ui/Ui.java:
```diff
@@ -0,0 +1,73 @@
+package noosy.ui;
```
Easy to understand names.Good job

rubinnn commented on src/main/java/noosy/ui/Ui.java:
```diff
@@ -0,0 +1,73 @@
+        return scanner.nextLine();
+    }
+
+    public void showLine() {
+        System.out.println("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
+    }
```
Good job extracting the lines out to make code cleaner

rubinnn reviewed (commented):
LGTM.Just some nits to fix.Well done.

potaotototo committed:
Add JUunit tests

potaotototo committed:
Merge branch-A-JUnit into master

potaotototo committed:
Create Jar file

potaotototo committed:
Merge branch-A-Jar into master

potaotototo committed:
Add JavaDoc

potaotototo committed:
Merge branch-A-JavaDoc into master

potaotototo committed:
Reorder import statements to comply by coding standard

potaotototo committed:
Add FindCommand for user to find tasks with a keyword

potaotototo committed:
Merge branch-A-CodingStandard into master

potaotototo committed:
Merge branch-Level-9 into master

potaotototo committed:
Add checkstyle

potaotototo committed:
Merge branch-A-CheckStyle into master

potaotototo committed:
Add css files

XinghuaJulia commented on src/main/java/noosy/task/Deadline.java:
```diff
@@ -0,0 +1,82 @@
+     */
+    @Override
+    public String toString() {
+        String formattedDate = (due != null) ? due.format(OUTPUT_FORMATTER) : "Date can't be processed by noosy.Noosy";
```
Perhaps can condense the "Date can't be processed by noosy.Noosy" into a noosyWrongDateFormatException? It seems that there are quite a few repeats...

XinghuaJulia commented on src/main/java/noosy/parser/Parser.java:
```diff
@@ -0,0 +1,139 @@
+                        "deadline – add a task with a deadline \n" +
+                        "event – add a task with duration \n" +
+                        "on – check tasks on a certain date \n" +
+                        "bye – say goodbye to Noosy \n");
```
Your parse() method is very very very long XD, consider separating each of the cases into its own method...

XinghuaJulia commented on README.md:
```diff
@@ -1,6 +1,6 @@
-# Duke project template
+# noosy.Noosy
```
Cute name and cute concept, good job :D

XinghuaJulia reviewed (commented):
LGTM, the code is mostly clean... with minor nits here and there

potaotototo committed:
Merge branch-Level-10 into master

potaotototo committed:
Try adding welcome message but failed

potaotototo committed:
Add assertions for Command subclasses, Noosy, Storage and Parser files

potaotototo committed:
Improve code quality

potaotototo committed:
Find tasks using Stream API

potaotototo committed:
Merge pull request #2 from potaotototo/branch-A-Assertions

Add assertions for Command subclasses, Noosy, Storage and Parser files

potaotototo committed:
Update code quality

potaotototo committed:
Merge branch-A-CodeQuality into master

potaotototo committed:
Merge pull request #4 from potaotototo/branch-A-CodeQuality

Improve code quality

potaotototo committed:
Merge branch 'master' of https://github.com/potaotototo/ip

potaotototo committed:
Resolve merge conflict

potaotototo committed:
Merge pull request #3 from potaotototo/branch-A-Streams

Find tasks using Stream API

potaotototo committed:
Merge branch 'master' of https://github.com/potaotototo/ip
Merge all three branches

potaotototo committed:
Deal with duplicate task description

potaotototo committed:
Merge branch-C-DetectDuplicates into master

potaotototo committed:
Ensure consistent date formatting in task storage

Date was previously all formatted as Output format.

This affects readability as the date is inconsistent.

Reformatted date to be stored in file with INPUT_FORMATTER.

This ensures consistency in data storage.

potaotototo committed:
Add test cases for NoosyDateException and parseEvent method

- Added a unit test to validate NoosyDateException is thrown when an invalid date is provided in the parseDeadline method.
- Added tests for the parseEvent method:
  - Successfully parses valid input with correct date and time format.
  - Throws NoosyException when the duration is incomplete (missing /from or /to).
  - Throws NoosyDateTimeException when the provided date or time is in an invalid format.

These tests improve the robustness of the Parser class by ensuring error handling is properly implemented for date and time-related inputs.

potaotototo committed:
Finalise formatting changes

potaotototo committed:
Add title

potaotototo committed:
Add README file

potaotototo committed:
Update screenshot in user guide

potaotototo committed:
Update README.md

potaotototo committed:
Update README.md
